src/values/AbstractObjectValue.js

```diff
@@ -149,6 +170,25 @@ export default class AbstractObjectValue extends AbstractValue {
     this.cachedIsSimpleObject = true;
   }
 
+  // Use this only if it is known that only the string properties of the snapshot will be accessed.
+  getSnapshot(): AbstractObjectValue {
+    if (this.isIntrinsic()) return this; // already temporal
+    if (this.values.isTop()) return this; // always the same
+    if (this.kind === "conditional") {
+      let [c, l, r] = this.args;
+      invariant(l instanceof ObjectValue || l instanceof AbstractObjectValue);
+      let ls = l.getSnapshot();
+      invariant(r instanceof ObjectValue || r instanceof AbstractObjectValue);
+      let rs = r.getSnapshot();
+      invariant(c instanceof AbstractValue);
+      let absVal = AbstractValue.createFromConditionalOp(this.$Realm, c, ls, rs, this.expressionLocation);
+      invariant(absVal instanceof AbstractObjectValue);
+      return absVal;
+    }
+    this.makeFinal();
```

Comment:
(Are there any other cases where we'd want to do a deep getSnapshot traversal?)

Reply:
None comes to mind. This is conservative, so we can afford to find out by running over real code.